### PR TITLE
[Major] Pipelines usage changed.

### PR DESCRIPTION
**Pull request by atb00ker**
**Problem:** The default pipeline class of scrapy has limitation, specifically the instance cannot be accessed from inside the spider.

**Solution:** In order to tackle that problem i have made a new class, which acts as the pipeline class but has the instance stored in the spider.

**Why am i tagged for review?** 
Because moving out from the default class changes the functionality of the pipelines,
hence, changes to all the spiders is made and everyone involved needs to understand the changes to implement the currently available APIs and the APIs of this class that will be made in the future!

**Recommended Action:** Read your spider's code and pipelines code.

### CODE REVIEW

**Comment by ATB00ker** [comment]
@omi10859 , @anujagrazzel; Have a look at this PR! :)

**Comment by parthsharma2** [comment]
I think you need to add a `yield` before each `self.postgres.process_item(item, self.name)` in each spider.

**Comment by atb00ker** [comment]
**Update: ** Using a spider variable variable for solving the instance issue, no change is needed in the spiders, everything is ensured by pipelines.
**Action Needed: ** Please read the pipelines file, It looks good to me and all the spiders run, but a second eye on major changes is always welcomed! :)
@parthsharma2, @thisisayush, @vipulgupta2048, @anujagrazzel, @omi10859

File: scrapeNews/scrapeNews/pipelines.py [hunk 1]
```diff
@@ -78,93 +100,51 @@ def open_spider(self, spider):
...
-                    self.recordedArticles += 1
-                except Exception as Error:
-                    loggerError.error(str(Error) + " occured at " + str(item.get('link')))
-                finally:
-                    return item
+            if (self.connection.status != 1):
+                self.openConnection()
+            postgresQuery = "INSERT INTO " + os.environ['NEWS_TABLE'] + " (title, content, image, link, newsDate, site_id, datescraped) VALUES (%s, %s, %s, %s, %s, %s, NOW())"
+            processedDate = str(parser.parse(item.get('newsDate'), ignoretz=False))
+            self.cursor.execute(postgresQuery,
```
Comment (thisisayush) [inline scrapeNews/scrapeNews/pipelines.py]: Implement None or Null Checks before executing this.

File: scrapeNews/scrapeNews/spiders/asianage.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def getPageTitle(self, newsBox):
 
     def getPageLink(self, newsBox):
         data = 'http://www.asianage.com/newsmakers' + newsBox.xpath('div/a/@href').extract_first()
-        if (data is None):
+        if (data == 'http://www.asianage.com/newsmakers'):
             loggerError.error(newsBox)
```
Comment (thisisayush) [inline scrapeNews/scrapeNews/spiders/asianage.py]: Improve error Logging! How are you supposed to know what data was missing!

File: scrapeNews/scrapeNews/pipelines.py (hunk 1 above)
Comment (atb00ker) [reply scrapeNews/scrapeNews/pipelines.py]: None or Null will automatically be rejected by the database, that's why i removed it!

File: scrapeNews/scrapeNews/spiders/asianage.py (hunk 2 above)
Comment (atb00ker) [reply scrapeNews/scrapeNews/spiders/asianage.py]: Update for logging is under development! One PR must contain only one logical change :)

**Pull request merged**